code/controllers/subsystem/discord.dm

```diff
@@ -81,70 +91,184 @@ SUBSYSTEM_DEF(discord)
+  * This gets the most recent entry from the discord link table that is associated with the given ckey
+  *
+  * Arguments:
+  * * lookup_ckey A string representing the ckey to search on
+  */
 /datum/controller/subsystem/discord/proc/lookup_id(lookup_ckey)
-	//We cast the discord ID to varchar to prevent BYOND mangling
-	//it into it's scientific notation
-	var/datum/db_query/query_get_discord_id = SSdbcore.NewQuery(
-		"SELECT CAST(discord_id AS CHAR(25)) FROM [format_table_name("player")] WHERE ckey = :ckey",
-		list("ckey" = lookup_ckey)
-	)
-	if(!query_get_discord_id.Execute())
-		qdel(query_get_discord_id)
-		return
-	if(query_get_discord_id.NextRow())
-		. = query_get_discord_id.item[1]
-	qdel(query_get_discord_id)
+	var/datum/discord_link_record/link = find_discord_link_by_ckey(lookup_ckey)
+	if(link)
+		return link.discord_id
 
-// Returns ckey from ID
+/**
+  * Given a discord id as a string, look up the ckey attached to that account, if any
+  *
+  * This gets the most recent entry from the discord_link table that is associated with this discord id snowflake
+  *
+  * Arguments:
+  * * lookup_id The discord id as a string
+  */
 /datum/controller/subsystem/discord/proc/lookup_ckey(lookup_id)
-	var/datum/db_query/query_get_discord_ckey = SSdbcore.NewQuery(
-		"SELECT ckey FROM [format_table_name("player")] WHERE discord_id = :discord_id",
-		list("discord_id" = lookup_id)
+	var/datum/discord_link_record/link  = find_discord_link_by_discord_id(lookup_id)
+	if(link)
+		return link.ckey
+
+/datum/controller/subsystem/discord/proc/get_or_generate_one_time_token_for_ckey(ckey)
+	// Is there an existing valid one time token
+	var/datum/discord_link_record/link = find_discord_link_by_ckey(ckey, timebound = TRUE)
+	if(link)
+		return link.one_time_token
+
+	// Otherwise we make one
+	return generate_one_time_token(ckey)
+
+/**
+  * Generate a timebound token for discord verification
+  *
+  * This uses the common word list to generate a six word random token, this token can then be fed to a discord bot that has access
+  * to the same database, and it can use it to link a ckey to a discord id, with minimal user effort
+  *
+  * It returns the token to the calling proc, after inserting an entry into the discord_link table of the following form
+  *
+  * ```
+  * (unique_id, ckey, null, the current time, the one time token generated)
+  * the null value will be filled out with the discord id by the integrated discord bot when a user verifies
+  * ```
+  *
+  * Notes:
+  * * The token is guaranteed to unique during it's validity period
+  * * The validity period is currently set at 4 hours
+  * * a token may not be unique outside it's validity window (to reduce conflicts)
+  *
+  * Arguments:
+  * * ckey_for a string representing the ckey this token is for
+  *
+  * Returns a string representing the one time token
+  */
+/datum/controller/subsystem/discord/proc/generate_one_time_token(ckey_for)
+
+	var not_unique = TRUE
+	var one_time_token = ""
```

Comment:
```suggestion
	var/not_unique = TRUE
	var/one_time_token = ""
```

Reply:
Though it works, we don't use this syntax on DM (the only instances we have are js parts  of it)
It's in the contributing guidelines as absolute pathing.